tensorboard/components/tf_data_selector/tf-data-select-row.ts

```diff
@@ -14,64 +14,189 @@ limitations under the License.
 ==============================================================================*/
 namespace tf_data_selector {
 
+enum Type {
+  RUN = 1,
+  TAG,
+}
+
 Polymer({
   is: 'tf-data-select-row',
   properties: {
-    experiment: Object,
+    experiment: {
+      type: Object,
+      value: () => ({
+        id: null,
+        name: 'Unknown experiment',
+        startTime: null,
+      }),
+    },
+
+    // Required field.
+    persistenceNumber: Number,
 
-    _runs: Array,
+    noExperiment: {
+      type: Boolean,
+      value: false,
+    },
 
-    selectedRuns: {
+    _runs: {
       type: Array,
-      notify: true,
       value: () => [],
     },
 
-    _runSelectionState: {
-      type: Object,
-      observer: '_storeRunSelectionState',
-      value: () => tf_storage.getObject('runSelectionState') || {},
+    _tags: {
+      type: Array,
+      value: () => [],
+    },
+
+    selection: {
+      type: Array,
+      notify: true,
+      computed: '_computeSelection(_selectedRuns, _selectedTags)',
     },
 
-    _runRegexInput: {
-      type: String,
-      value: tf_storage.getStringInitializer('regexInput', {
-            defaultValue: '',
-            polymerProperty: '_runRegexInput',
-          }),
-      observer: '_storeRunRegexInput',
+    _runSelectionStateString: {type: String, value: ''},
+
+    _selectedRuns: {
+      type: Array,
+      value: () => [],
+    },
+
+    _tagSelectionStateString: {type: String, value: ''},
+
+    _selectedTags: {
+      type: Array,
+      value: () => [],
     },
   },
 
-  get _runsStore() {
-    return tf_backend.runsStore;
+  observers: [
+    '_persistSelectedRuns(_selectedRuns)',
+    '_persistSelectedTags(_selectedTags)',
+  ],
+
+  _getPersistenceKey(type: Type): string {
+    const number = this.persistenceNumber || 0;
+    switch (type) {
+      case Type.RUN:
+        return `g${number}r`;
```

Comment:
Does the "g" prefix have any particular meaning?  "group"?  Maybe good to specify in a comment or something to make it easy to remember as being a reserved URL param prefix.